doc/topics/core-structure.md

```diff
@@ -0,0 +1,202 @@
+# Node Core's Internal Structure
+
+This document covers the basics of how Node.js is structured, in an attempt to
+make it easier to contribute to.
+
+Note: The Node.js codebase is in a constant amount of flux, things can and will
+change over time, and so also parts of this document will go out of date over
+time.
+
+
+## Top-Level File & Folder Structure
+
+At the root of Node.js you have a set of files and folders that can be roughly
+identified as the following:
+
+- CAPITALS documentation files
+- Documentation (`doc/`)
+- Top-level Build Tooling
+- JavaScript "standard" Library (`lib/`)
+- C++ "native" Source (`src/`)
+- Tests & Benchmarks (`test/` & `benchmark/`)
+- Internal / Build Tooling (`tools/`)
+- Dependencies (`deps/`)
+- .dotfiles
```

Comment:
/doc/changelogs/*.md
On May 23, 2016 7:45 AM, "Jeremiah Senkpiel" <notifications@github.com>
wrote:

> In doc/topics/core-structure.md
> <https://github.com/nodejs/node/pull/6885#discussion_r64234969>:
>
> > +
> > +
> > +## Top-Level File & Folder Structure
> > +
> > +At the root of Node.js you have a set of files and folders that can be roughly
> > +identified as the following:
> > +
> > +- CAPITALS documentation files
> > +- Documentation (`doc/`)
> > +- Top-level Build Tooling
> > +- JavaScript "standard" Library (`lib/`)
> > +- C++ "native" Source (`src/`)
> > +- Tests & Benchmarks (`test/` & `benchmark/`)
> > +- Internal / Build Tooling (`tools/`)
> > +- Dependencies (`deps/`)
> > +- .dotfiles
>
> CHANGELOG?
>
> Huh? That's a "CAPITALS" file.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/node/pull/6885/files/45979673c36c8d2e485bcd06617042703f0fa5e6#r64234969>
>